src/main/java/teammates/common/datatransfer/InstructorAttributes.java

```diff
@@ -17,6 +17,9 @@
  * The data transfer class for Instructor entities.
  */
 public class InstructorAttributes extends EntityAttributes {
+    // Note: we should disable HTML escaping when creating gson string of the InstructorAttributes
+    // objects because if the name of the instructor contains any html character, we cannot
+    // search for the object by the name anymore.
```

Comment:
We have discussed about it once here https://github.com/TEAMMATES/teammates/pull/5058#discussion-diff-56760084
The problem is that the old gson configuration uses a different sanitizer from ours. Therefore we cannot recover sanitized strings using our current function in `StringHelper`.